### PR TITLE
implement SendableBuilderImpl

### PR DESCRIPTION
**Pull request by ariovistus**

### CODE REVIEW

File: wpilib/wpilib/sendablebuilder.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    def addBooleanProperty(self, key, getter, setter):
+        """
+        Add a boolean property.
+   
+        :param key:     property name
+        :type key: str
+        :param getter:  getter function (returns current value)
+        :type getter: typing.Callable[[], bool]
+        :param setter:  setter function (sets new value)
+        :type setter: typing.Callable[[networktables.entry.NetworkTableEntry, str, bool, Any], None]
```
Comment (ariovistus) [inline wpilib/wpilib/sendablebuilder.py]: need guidance on the right way to express these

Comment (auscompgeek) [reply wpilib/wpilib/sendablebuilder.py]: the right way = not in the docstring

Comment (ariovistus) [reply wpilib/wpilib/sendablebuilder.py]: I though 3.4 compat prevents it from going elsewhere

Comment (virtuald) [reply wpilib/wpilib/sendablebuilder.py]: Is there a compelling reason to support 3.4 still?

Comment (auscompgeek) [reply wpilib/wpilib/sendablebuilder.py]: there's a backport of the typing module in either case

Comment (virtuald) [reply wpilib/wpilib/sendablebuilder.py]: Honestly, I would say pick one of the two:

* Use the type annotations
* Build the documentation locally, and change the type to match what you would want it to look like as sphinx renders it. I'm not too picky about it... but it seems like this wouldn't render very well?

File: wpilib/wpilib/sendablebuilder.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        """
+        self.table = table
+
+    def getTable(self):
+        """
+        Get the network table.
+
+        :returns: The network table
+        :rtype: networktables.networktable.NetworkTable
+        :rtype: :class:`networktables.networktable.NetworkTable`
```
Comment (auscompgeek) [inline wpilib/wpilib/sendablebuilder.py]: …

File: wpilib/wpilib/sendablebuilder.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        state.  This is called when entering and exiting Live Window mode.
+
+        :param func:    function
+        """
+        self.safeState = func
+
+    def setUpdateTable(self, func):
+        """
+         Set the function that should be called to update the network table
+        for things other than properties.  Note this function is not passed
```
Comment (auscompgeek) [inline wpilib/wpilib/sendablebuilder.py]: Indentation doesn't match.

File: wpilib/wpilib/sendablebuilder.py (hunk 1 above)
Comment (ariovistus) [reply wpilib/wpilib/sendablebuilder.py]: wow, typing generates some bizarre errors in places I didn't touch. guess we'll have to go with option 2. how do I build the documentation?

**Comment by ariovistus** [comment]
setter signature is wrong

File: wpilib/wpilib/sendablebuilder.py (hunk 1 above)
Comment (virtuald) [reply wpilib/wpilib/sendablebuilder.py]: Sorry, missed your question. Install sphinx and sphinx-rtd-theme, and do `make html`. If you're on Windows, not sure if that will actually work however.

**Pull request merged**